src/sentry/static/sentry/app/views/releases/list/projectList.tsx

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import styled from '@emotion/styled';
+
+import {tct} from 'app/locale';
+import ProjectBadge from 'app/components/idBadge/projectBadge';
+import {Project} from 'app/types';
+import space from 'app/styles/space';
+import DropdownAutoComplete from 'app/components/dropdownAutoComplete';
+import {AutoCompleteItem} from 'app/components/dropdownAutoCompleteMenu';
+
+type Props = {
+  projects: Project[];
+  maxLines?: number;
+};
+
+const ProjectList = ({projects, maxLines = 2}: Props) => {
+  let visibleProjects: Project[], hiddenProjects: Project[];
+
+  if (projects.length <= maxLines) {
+    visibleProjects = projects;
+    hiddenProjects = [];
+  } else {
+    // because we need one line for `and X more`
+    visibleProjects = projects.slice(0, maxLines - 1);
+    hiddenProjects = projects.slice(maxLines - 1, projects.length);
+  }
+
+  return (
+    <React.Fragment>
+      {visibleProjects.map(project => (
+        <StyledProjectBadge project={project} avatarSize={14} key={project.slug} />
+      ))}
+      {hiddenProjects.length > 0 && (
+        <StyledDropdownAutoComplete
```

Comment:
What does this autocomplete do after filtering?

Reply:
Nothing, it's just for displaying. Some orgs have apparently as many as 40 projects belonging to one release, so this should help scroll/search through them.

Reply:
Everywhere else we use an autocomplete it is for selecting values. This would be a one-off where an autocomplete isn't being used for section. Could we put the 'other' values in something like a hovercard?

Reply:
Yeah I would expect this kind of UI pattern to be used when there's additional behavior after filtering e.g. in the column header to filter rows with a project.

Reply:
You guys are right. Thanks to @markstory and @mikeclarke  we found out, that only 2% of releases have more than 2 projects (max is 96!). So I changed it a bit - `and X more` now links to release detail page, where you can find all the projects + if the X is not too big, we also show a hovercard with the rest of the projects listed right there. Bonus points for using recursion 😁

![chrome-capture (4)](https://user-images.githubusercontent.com/9060071/73972612-68401500-4921-11ea-8a4d-8ff23e5ead8b.gif)

Reply:
What happens if the X is too big?

Reply:
There will be a link to release detail page saying ‘and 95 more’ with no hovercard.